rockchip: add FriendlyARM NanoPC T4 support

msdos03 opened the pull request:
Hardware
--------
RockChip RK3399 ARM64 (6 cores)
4GB LPDDR3 RAM
1x 1000 Base-T
1 GPIO LED (status)
HDMI 2.0
3.5mm TRRS AV jack
Micro-SD slot
16GB eMMC
1x USB 3.0 Port
2x USB 2.0 Port
1x USB Type-C Port
1x M.2 PCI-E Port
AP6356S (BCM4356) SDIO WiFi & Bluetooth adapter

msdos03 commented:
@1715173329

1715173329 commented:
https://github.com/openwrt/openwrt/blob/e9911f10e482f3174f745a36c0c9fd7964758caf/target/linux/rockchip/patches-6.1/107-arm64-dts-rockchip-Update-LED-properties-for-Orange-.patch
it would be nice to add system-LED aliases for OpenWrt, otherwise LGTM

msdos03 commented:
@1715173329 Done. Is that OK?

1715173329 commented:
> @1715173329 Done. Is that OK now?

Please split your changes into 2 commits, one for the uboot and another for the target.
Otherwise LGTM

msdos03 commented:
@1715173329 OK, what else should be done before merge?

1715173329 commented:
> @1715173329 OK, what else should be done before merge?

Just take a rest ;)

msdos03 committed:
uboot-rockchip: add FriendlyARM NanoPC T4 support

Add support for the FriendlyARM NanoPC T4.
Reduce memory frequency to avoid random crash.

Signed-off-by: Lu jicong <jiconglu58@gmail.com>

msdos03 committed:
rockchip: add FriendlyARM NanoPC T4 support

Hardware
--------
RockChip RK3399 ARM64 (6 cores)
4GB LPDDR3 RAM
1x 1000 Base-T
1 GPIO LED (status)
HDMI 2.0
3.5mm TRRS AV jack
Micro-SD slot
16GB eMMC
1x USB 3.0 Port
2x USB 2.0 Port
1x USB Type-C Port
1x M.2 PCI-E Port
AP6356S (BCM4356) SDIO WiFi & Bluetooth adapter
--------
Note: AP6356S is not supported yet due to the lack of firmware and NVRAM

Signed-off-by: Lu jicong <jiconglu58@gmail.com>

msdos03 committed:
linux-firmware: broadcom: package bcm4356 NVRAM for NanoPC T4

Prepare for NanoPC T4 WIFI support.

Signed-off-by: Lu jicong <jiconglu58@gmail.com>

msdos03 committed:
rockchip: enable wifi support for NanoPC T4

wifi module: AP6356S (BCM4356) SDIO

Signed-off-by: Lu jicong <jiconglu58@gmail.com>

msdos03 commented:
@hauke

hauke commented on package/boot/uboot-rockchip/patches/105-nanopc-t4-fix-memory-unstability.patch:
```diff
@@ -0,0 +1,24 @@
+diff --git a/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi b/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi
+index 17201bcf41..8b6c9059ab 100644
+--- a/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi
++++ b/arch/arm/dts/rk3399-nanopc-t4-u-boot.dtsi
```
Please send this patch to the upstream U-Boot mailling list too to get it integrated into upstream U-Boot.

1715173329 replied:
https://lore.kernel.org/u-boot/20230705135816.182356-1-jiconglu58@gmail.com/

It's already there ;)
